### PR TITLE
rename serverFetch to externalFetch (solve #2101)

### PR DESCRIPTION
**Pull request by Mlocik97**
PR for: https://github.com/sveltejs/kit/issues/2101

---

### Before submitting the PR, please make sure you do the following
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by Mlocik97** [comment]
In CLI I'm getting this error, when running `pnpm test`

```
vite v2.4.3 building SSR bundle for production...
✓ 44 modules transformed.
.svelte-kit/output/server/app.js   61.09kb
• • • • • • • • • • • •   (12 / 12)

  Total:     460
  Passed:    439
  Skipped:   10
  Duration:  122637.21ms

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @sveltejs/kit@1.0.0-next.140 test:integration: `uvu test test.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @sveltejs/kit@1.0.0-next.140 test:integration script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mlocik97/.npm/_logs/2021-08-05T10_43_25_157Z-debug.log
/mnt/sda1/Programovanie/kit/packages/kit:
 ERROR  @sveltejs/kit@1.0.0-next.140 test: `npm run test:unit && npm run test:integration`
Exit status 1
 ERROR  Test failed. See above for more details.
```

**Comment by stalkerg** [comment]
I think this PR shouldn't contain `pnpm-lock.yaml`

File: documentation/docs/04-hooks.md [hunk 1]
```diff
@@ -92,19 +92,19 @@ export function getSession(request) {
...
-### serverFetch
+### externalFetch
 
 This function allows you to modify (or replace) a `fetch` request for an **external resource** that happens inside a `load` function that runs on the server (or during pre-rendering).
 
 For example, your `load` function might make a request to a public URL like `https://api.yourapp.com` when the user performs a client-side navigation to the respective page, but during SSR it might make sense to hit the API directly (bypassing whatever proxies and load balancers sit between it and the public internet).
 
 ```ts
-type ServerFetch = (req: Request) => Promise<Response>;
+type externalFetch = (req: Request) => Promise<Response>;
```
Comment (benmccann) [inline documentation/docs/04-hooks.md:102]: the type should still be capitalized

**Comment by Mlocik97** [comment]
> I think this PR shouldn't contain `pnpm-lock.yaml`

that file was added when I run command `pnpm test`, don't know why that file changed.

**Comment by benmccann** [comment]
for the pnpm lock file, make sure you're rebased against the latest version of `master` and are using the latest version of `pnpm`. I just confirmed that `pnpm install` doesn't update the lock file for me, so I agree ti shouldn't be updated in this PR

File: packages/kit/src/core/dev/index.js [hunk 2]
```diff
@@ -346,7 +346,7 @@ async function create_handler(vite, config, dir, cwd, manifest) {
 						hooks: {
 							getSession: hooks.getSession || (() => ({})),
 							handle: hooks.handle || (({ request, resolve }) => resolve(request)),
-							serverFetch: hooks.serverFetch || fetch
+							externalFetch: hooks.externalFetch || fetch
```
Comment (benmccann) [inline packages/kit/src/core/dev/index.js:349]: on line ~293 we have a check for old hooks. we should add a check for `serverFetch` as well

```
				if (/** @type {any} */ (hooks).getContext) {
					// TODO remove this for 1.0
					throw new Error(
						'The getContext hook has been removed. See https://kit.svelte.dev/docs#hooks'
					);
				}
```

**Comment by Mlocik97** [comment]
I fucked my branch, so I created new PR,...

**Pull request closed without merging**